src/core/Refiner.ml

```diff
@@ -78,6 +79,22 @@ struct
     () |> RM.emit (RefineEnv.location env) @@ fun fmt () ->
     Format.fprintf fmt "Emitted hole:@,  @[<v>%a@]@." (S.pp_sequent ~lbl ctx) tp
 
+  let print_boundary_warning lbl tm tp phi clo : unit m =
```

Comment:
Maybe we can make this provide some feedback like "The boundary is good!!" if it does match? That way the user will know to get rid of the shed.